File: app/src/main/java/com/dream/bjst/repayment/ui/RepaymentFragment.java
package com.dream.bjst.repayment.ui;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.annotation.NonNull;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.RecyclerView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.listener.OnItemChildClickListener;
import com.chad.library.adapter.base.listener.OnItemClickListener;
import com.dream.bjst.R;
import com.dream.bjst.app.MyAppKt;
import com.dream.bjst.base.BaseFragment;
import com.dream.bjst.bean.OverduedBean;
import com.dream.bjst.bean.RepaymentBean;
import com.dream.bjst.repayment.adapter.EmptyAdapter;
import com.dream.bjst.repayment.adapter.RePaymentAdapter;
import com.dream.bjst.repayment.adapter.RePaymentInAdapter;
import com.dream.bjst.utils.StatusBarUtils;

import java.util.ArrayList;
import java.util.List;

public class RepaymentFragment extends BaseFragment {
    RecyclerView mRecyclerView;
    private RePaymentAdapter mRePaymentAdapter;
    private RePaymentInAdapter mRePaymentInAdapter;
    private List<RepaymentBean> mList;

    @Override
    protected int setLayout() {
        //调整状态栏的字体颜色为黑色
        new StatusBarUtils().adjustWindow(getActivity(), true);
        return R.layout.fragment_repayment;
    }

    @Override
    protected void initView() {
        mRecyclerView = fvbi(R.id.repayment_rv_plan);
    }

    @Override
    protected void initData() {
        mRePaymentAdapter = new RePaymentAdapter();
        mRePaymentInAdapter=new RePaymentInAdapter();
        // 显示空布局
        mRePaymentAdapter.setEmptyView(getEmptyView());
        mRecyclerView.setAdapter(mRePaymentAdapter);
        setData();
    }

    private void setData() {
        mList = new ArrayList<>();
        List<OverduedBean> inList = new ArrayList<>();
        for (int i = 0; i < 3; i++) {
            inList.add(new OverduedBean("Overdued"));
        }
        for (int i = 0; i < 10; i++) {
            mList.add(new RepaymentBean(inList));
        }
        mRePaymentAdapter.setList(mList);
        mRePaymentInAdapter.setList(inList);

        //点击时间
        event();
    }

    private void event() {
        mRePaymentInAdapter.setOnItemChildClickListener(new OnItemChildClickListener() {
            @Override
            public void onItemChildClick(@NonNull BaseQuickAdapter adapter, @NonNull View view, int position) {
                startActivity(new Intent(MyAppKt.getMApplication(),RepaymentDetailActivity.class));
            }
        });

        mRePaymentAdapter.setOnItemClickListener(new OnItemClickListener() {
            @Override
            public void onItemClick(@NonNull BaseQuickAdapter<?, ?> adapter, @NonNull View view, int position) {
                startActivity(new Intent(MyAppKt.getMApplication(),RepaymentDetailActivity.class));
            }
        });
    }

    @Override
    public void onClick(View v) {

    }

    /**
     * 获取空布局
     */
    private View getEmptyView() {
        View view = LayoutInflater.from(mContext).inflate(R.layout.item_repayment_empty, null, false);
        Button btnLoadData = view.findViewById(R.id.btn_load_data);
        btnLoadData.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Navigation.findNavController(v).navigate(R.id.repayment_to_navigation_loan);
            }
        });
        return view;
    }
}

File: app/src/main/java/com/dream/bjst/repayment/ui/RepaymentFragment.kt
package com.dream.bjst.repayment.ui

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.core.view.isVisible
import androidx.navigation.Navigation
import com.blankj.utilcode.util.BarUtils
import com.blankj.utilcode.util.LogUtils
import com.blankj.utilcode.util.ToastUtils
import com.dream.bjst.R

import com.dream.bjst.databinding.FragmentRepaymentBinding
import com.dream.bjst.repayment.adapter.*
import com.dream.bjst.repayment.vm.RepaymentViewModel
import com.tcl.base.common.ui.BaseFragment
import com.tcl.base.kt.ktClick
import com.tcl.base.kt.ktSeriesClick
import com.tcl.base.kt.ktStartActivity
import rxhttp.wrapper.param.IFile

/**
 * 创建日期：2022-09-05 on 0:50
 * 描述:衣带渐宽终不悔、为伊消得人憔悴
 * 作者:HeGuiCun Administrator
 */
class RepaymentFragment : BaseFragment<RepaymentViewModel, FragmentRepaymentBinding>() {
    //数据适配器
    var overDueAdapter = OverDueAdapter()
    var dueTodayAdapter = DueTodayAdapter()
    var notDueAdapter = NotDueAdapter()

    override fun initView(savedInstanceState: Bundle?) {
        BarUtils.addMarginTopEqualStatusBarHeight(mBinding.repayment)
        initRv()
        viewModel.repaymentData()
        mBinding.smartRefresh.setOnRefreshListener {
            viewModel.repaymentData()
        }
    }

    /**
     * 数据更新
     */
    override fun startObserve() {
        super.startObserve()
        viewModel.repaymentResult.observe(this) {
            mBinding.smartRefresh.finishRefresh()
            mBinding.overdueRl.isVisible = !it.`9B829186908191BB8690918687`.isNullOrEmpty()
            if (!it.`9B829186908191BB8690918687`.isNullOrEmpty()) {
                overDueAdapter.setList(it.`9B829186908191BB8690918687`)
                mBinding.overDueNotice.text = "Daily penalty interest rate ${
                    it.`9B829186908191BB8690918687`[0].`9B829186908191BD9A809186918780A6958091`.toFloatOrNull()
                        ?.times(100)?.toUInt()
                } %"
            }
            mBinding.dueTodayRl.isVisible = !it.`908191A09B90958DBB8690918687`.isNullOrEmpty()
            if (!it.`908191A09B90958DBB8690918687`.isNullOrEmpty()) {
                dueTodayAdapter.setList(it.`908191A09B90958DBB8690918687`)
            }
            mBinding.notDueRl.isVisible = !it.`9A9B80B08191BB8690918687`.isNullOrEmpty()
            if (!it.`9A9B80B08191BB8690918687`.isNullOrEmpty()) {
                notDueAdapter.setList(it.`9A9B80B08191BB8690918687`)
            }

            val noData =
                overDueAdapter.data.isEmpty() && dueTodayAdapter.data.isEmpty() && notDueAdapter.data.isEmpty()
            mBinding.dataLay.isVisible = !noData
            mBinding.emptyLay.rootLay.isVisible = noData
        }
    }


    private fun initRv() {

        mBinding.overDueRv.adapter = overDueAdapter
        mBinding.dueTodayRv.adapter = dueTodayAdapter
        mBinding.notDueRv.adapter = notDueAdapter
        //点击事件
        event()
    }

    private fun event() {
        mBinding.emptyLay.btnLoadData.ktClick {
            Navigation.findNavController(mBinding.emptyLay.rootLay)
                .navigate(R.id.repayment_to_navigation_loan)
        }
        overDueAdapter.setOnItemChildClickListener { adapter, view, position ->
            when (view.id) {
                R.id.settleNowBtn ->
                    ktStartActivity(RepaymentDetailActivity::class) {
                        putExtra("detailId", overDueAdapter.data[position].`969B86869B83BD90`)
                    }
            }
        }
        notDueAdapter.setOnItemChildClickListener { adapter, view, position ->
            when (view.id) {
                R.id.settleNowBtn -> ktStartActivity(RepaymentDetailActivity::class) {
                    putExtra("detailId", notDueAdapter.data[position].`969B86869B83BD90`)
                }
            }

        }
        dueTodayAdapter.setOnItemChildClickListener { adapter, view, position ->
            when (view.id) {
                R.id.settleNowBtn -> ktStartActivity(RepaymentDetailActivity::class) {
                    putExtra("detailId", dueTodayAdapter.data[position].`969B86869B83BD90`)
                }
            }

        }


    }
}